src/hostapi/wdmks/pa_win_wdmks.c

```diff
@@ -6582,31 +6582,41 @@ static PaError PaPinRenderSubmitHandler_WaveCyclic(PaProcessThreadInfo* pInfo, u
 static PaError PaPinCaptureEventHandler_WaveRTEvent(PaProcessThreadInfo* pInfo, unsigned eventIndex)
 {
     unsigned long pos;
-    unsigned realInBuf;
-    unsigned frameCount;
+    unsigned frameCount = 0;
+    unsigned bytesToRead;
     PaWinWdmIOInfo* pCapture = &pInfo->stream->capture;
     const unsigned halfInputBuffer = pCapture->hostBufferSize >> 1;
     PaWinWdmPin* pin = pCapture->pPin;
     DATAPACKET* packet = 0;
 
     /* Get hold of current ADC position */
     pin->fnAudioPosition(pin, &pos);
-    /* Wrap it (robi: why not use hw latency compensation here ?? because pos then gets _way_ off from
-    where it should be, i.e. at beginning or half buffer position. Why? No idea.)  */
-
     pos %= pCapture->hostBufferSize;
-    /* Then realInBuf will point to "other" half of double buffer */
-    realInBuf = pos < halfInputBuffer ? 1U : 0U;
-
-    packet = pInfo->stream->capture.packets + realInBuf;
+    pos &= ~(pCapture->bytesPerFrame - 1);
```

Comment:
This only works if `bytesPerFrame` is a power of two. But we can have packed 24 bit data or 5 channels, so this won't work as-is. Do you think the following change is good?

  `pos -= pos % pCapture->bytesPerFrame;`

Reply:
well, 24-bit is packed into 32-bit containers by Windows. Regarding 5 channels, I am not sure if it is possible.